src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/MulticlassLogisticRegression.cs

```diff
@@ -19,25 +19,28 @@
 using Microsoft.ML.Trainers;
 using Newtonsoft.Json.Linq;
 
-[assembly: LoadableClass(typeof(LogisticRegressionMulticlassClassificationTrainer), typeof(LogisticRegressionMulticlassClassificationTrainer.Options),
+[assembly: LoadableClass(typeof(LbfgsMaximumEntropyTrainer), typeof(LbfgsMaximumEntropyTrainer.Options),
     new[] { typeof(SignatureMulticlassClassifierTrainer), typeof(SignatureTrainer) },
-    LogisticRegressionMulticlassClassificationTrainer.UserNameValue,
-    LogisticRegressionMulticlassClassificationTrainer.LoadNameValue,
+    LbfgsMaximumEntropyTrainer.UserNameValue,
+    LbfgsMaximumEntropyTrainer.LoadNameValue,
     "MulticlassLogisticRegressionPredictorNew",
-    LogisticRegressionMulticlassClassificationTrainer.ShortName,
+    LbfgsMaximumEntropyTrainer.ShortName,
     "multilr")]
 
-[assembly: LoadableClass(typeof(MulticlassLogisticRegressionModelParameters), null, typeof(SignatureLoadModel),
+[assembly: LoadableClass(typeof(MaximumEntropyModelParameters), null, typeof(SignatureLoadModel),
     "Multiclass LR Executor",
-    MulticlassLogisticRegressionModelParameters.LoaderSignature)]
+    MaximumEntropyModelParameters.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(void), typeof(LbfgsMaximumEntropyTrainer), null, typeof(SignatureEntryPointModule), LbfgsMaximumEntropyTrainer.LoadNameValue)]
 
 namespace Microsoft.ML.Trainers
 {
     /// <include file = 'doc.xml' path='doc/members/member[@name="LBFGS"]/*' />
     /// <include file = 'doc.xml' path='docs/members/example[@name="LogisticRegressionClassifier"]/*' />
-    public sealed class LogisticRegressionMulticlassClassificationTrainer : LbfgsTrainerBase<LogisticRegressionMulticlassClassificationTrainer.Options,
-        MulticlassPredictionTransformer<MulticlassLogisticRegressionModelParameters>, MulticlassLogisticRegressionModelParameters>
+    public sealed class LbfgsMaximumEntropyTrainer : LbfgsTrainerBase<LbfgsMaximumEntropyTrainer.Options,
```

Comment:
Can we have prefix match between the Trainer  and ModelParameter  class ?

Reply:
This is not the only trainer that will produce MaximumEntropyTrainer.. Please consider this trainer and the associated model separately. Otherwise, every trainer which produces linear model should be called LinearModelTrainer.

---
In reply to: [266683935](https://github.com/dotnet/machinelearning/pull/2976#discussion_r266683935) [](ancestors = 266683935)